### PR TITLE
Changes to adapt the code to landsat collection 2

### PR DESCRIPTION
**Pull request by pacs27**

### CODE REVIEW

File: etbrasil/geesebal/timeseries_bcj.py [hunk 1]
```diff
@@ -111,57 +117,105 @@ def retrieveETandMeteo(image):
...
                 z_alt = srtm.select('elevation')
                 slope = ee.Terrain.slope(z_alt)
 
                 #SPECTRAL IMAGES (NDVI, EVI, SAVI, LAI, T_LST, e_0, e_NB, long, lat)
                 image=fexp_spec_ind(image)
-
+                
                 #LAND SURFACE TEMPERATURE
+                # TODO: IS THIS CORRECTION NECESSARY? Answer: No, because it uses the brightness temperature
                 image=LST_DEM_correction(image, z_alt, T_air, UR,sun_elevation,_hour,_minutes)
```
Comment (bjonesneu) [inline etbrasil/geesebal/timeseries_bcj.py:186]: Does the comment on line 185 mean we should remove line 186?

File: etbrasil/geesebal/timeseries_bcj.py [hunk 2]
```diff
@@ -262,24 +326,28 @@ def extractMaxValue(var):
...
+                    'ET_R_max':     ET_max_point_get,
                     'NDVI': NDVI_point_get,
                     'AirT_G': T_air_point_get,
                     'LandT_G': T_land_point_get,
                     'ux': ux_point_get,
                     'UR': UR_point_get,
                     'z_alt': z_alt_point_get,
                     'slope': slope_point_get,
-                    'precip': precip
+                    'precip': 1
```
Comment (bjonesneu) [inline etbrasil/geesebal/timeseries_bcj.py:337]: Why was this changed to a constant of 1?  We're using the precip value retrieved from a functional call in this class in a downstream task.

File: etbrasil/geesebal/timeseries_bcj.py [hunk 3]
```diff
@@ -111,57 +117,105 @@ def retrieveETandMeteo(image):
...
+                'LandT_G': None,
+                'ux': None,
+                'UR': None,
+                'z_alt': None,
+                'slope': None,
+                'precip': None
+                
             })
             errmsg = None
+            image_bands_max = image.reduceRegion(
```
Comment (bjonesneu) [inline etbrasil/geesebal/timeseries_bcj.py:135]: What are lines 135-150 adding or doing differently in the algorithm?

File: etbrasil/geesebal/timeseries_bcj.py [hunk 4]
```diff
@@ -111,57 +117,105 @@ def retrieveETandMeteo(image):
...
+                
+                
+                
                 #COLD PIXEL
                 d_cold_pixel=fexp_cold_pixel(image, geometryReducer, p_top_NDVI, p_coldest_Ts)
 
                 #COLD PIXEL NUMBER
                 n_Ts_cold = ee.Number(d_cold_pixel.get('temp'))
-
+                image_bands_max = image.reduceRegion(
```
Comment (bjonesneu) [inline etbrasil/geesebal/timeseries_bcj.py:196]: how to lines 196-232 compare with 135-150 above?

File: etbrasil/geesebal/timeseries_bcj.py [hunk 5]
```diff
@@ -170,19 +224,28 @@ def retrieveETandMeteo(image):
...
                 image=fexp_sensible_heat_flux_bcj(image, ux, UR,Rn24hobs,n_Ts_cold,
                                             d_hot_pixel, date_string, geometryReducer)
 
                 #DAILY EVAPOTRANSPIRATION (ET_24H) [MM DAY-1]
                 image=fexp_et(image,Rn24hobs)
-
+                
                 NAME_FINAL=ee.String(LANDSAT_ID).slice(0,5).cat(ee.String(LANDSAT_ID).slice(10,17)).cat(ee.String(LANDSAT_ID).slice(17,25))
-
+                def checkIfCoordinateRepresentAMaskedPoint(image):
```
Comment (bjonesneu) [inline etbrasil/geesebal/timeseries_bcj.py:239]: what is this new functioning doing in the overall processing?

File: etbrasil/geesebal/timeseries_bcj.py [hunk 6]
```diff
@@ -297,28 +365,52 @@ def extractMaxValue(var):
...
-
-        fc8 = self.collection_l8.map(f_cloudMaskL8_SR) \
-                                .map(f_albedoL8) \
-                                .map(verifyMeteoAvail) \
-                                .filter(ee.Filter.gt('meteo_count', 0)) \
-                                .map(get_meteorology) \
-                                .map(retrieveETandMeteo)
+        # # .map(prepSrLandsat8) \
+        # # TODO: Try to remove this
+        # # Reduce the collection to a collection of booleans, where each boolean indicates whether the corresponding image has any `None` values.
```
Comment (bjonesneu) [inline etbrasil/geesebal/timeseries_bcj.py:407]: I'm unsure I understand this - what is the change in functionality to the prior algorithm and why?  Is it a recommendation to implement this?

File: etbrasil/geesebal/meteorology.py [hunk 7]
```diff
@@ -161,9 +161,19 @@ def get_meteorology(image):
...
     # ACTUAL VAPOR PRESSURE [KPA]
-    ea = tdp.expression(
-        '0.6108 * (exp((17.27 * T_air) / (T_air + 237.3)))',{
-        'T_air': tdp.subtract(273.15)})
+    #PRESSURE [PA] CONVERTED TO KPA
+    i_P_med= NEXT_IMAGE.select('surface_pressure').subtract(PREVIOUS_IMAGE.select('surface_pressure')).multiply(DELTA_TIME).add(PREVIOUS_IMAGE.select('surface_pressure')).divide(ee.Number(1000));
+
+    # SPECIFIC HUMIDITY [KG KG-1]
+    i_q_med =ee.ImageCollection(Constants.WEATHER_NASA_GLDAS_V021_NOAH_G025_T3H).select(['Qair_f_inst'],["specific_humidity"]).filterBounds(image.geometry()).filter(ee.Filter.date(ee.Date(time_start),ee.Date(time_start).advance(1,'day'))).first(); 
+    ea = i_P_med.expression('(1/0.622)*Q*P',{
```
Comment (bjonesneu) [inline etbrasil/geesebal/meteorology.py:169]: With this new version of ea calc, I can remove the old code entirely?

File: etbrasil/geesebal/tools.py [hunk 8]
```diff
@@ -1,691 +1,779 @@
...
-#LAB - HIDROLOGIA DE GRANDE ESCALA [HGE] website: https://www.ufrgs.br/hge/author/hge/
-#UNIVERSITY - UNIVERSIDADE FEDERAL DO RIO GRANDE DO SUL - UFRGS
-#RIO GRANDE DO SUL, BRAZIL
-
-#DOI
-#VERSION 0.1.1
-#CONTACT US: leonardo.laipelt@ufrgs.br
-
-#----------------------------------------------------------------------------------------#
+# ----------------------------------------------------------------------------------------#
```
Comment (bjonesneu) [inline etbrasil/geesebal/tools.py:1]: Its kind of difficult to see the overview of what changed due to the number of formatting changes made by the code editor.  Which algorithms changed substantially?

Comment (bjonesneu) [reply etbrasil/geesebal/tools.py:1]: Is there some overlap between this file and meteorology.py?  That may be from the legacy code.

File: etbrasil/geesebal/timeseries_bcj.py (hunk 1 above)
Comment (pacs27) [reply etbrasil/geesebal/timeseries_bcj.py:186]: Ignore the last part, it has nothing to do with brightness. As the data we use from LST is provided directly to us, I don't know if it is necessary to make this correction.  I think it is, but we have to look at it

File: etbrasil/geesebal/timeseries_bcj.py (hunk 2 above)
Comment (pacs27) [reply etbrasil/geesebal/timeseries_bcj.py:337]: Sorry, my mistake. It is difficult to debug when working with GEE and I changed the variable to see if it was giving an error

File: etbrasil/geesebal/timeseries_bcj.py (hunk 3 above)
Comment (pacs27) [reply etbrasil/geesebal/timeseries_bcj.py:135]: Nothing, they are mainly for debugging purposes. We can remove them

File: etbrasil/geesebal/timeseries_bcj.py (hunk 4 above)
Comment (pacs27) [reply etbrasil/geesebal/timeseries_bcj.py:196]: The same, they are mainly for debugging purposes. We can remove them

File: etbrasil/geesebal/timeseries_bcj.py (hunk 5 above)
Comment (pacs27) [reply etbrasil/geesebal/timeseries_bcj.py:239]: Nothing, you can remove it

File: etbrasil/geesebal/timeseries_bcj.py (hunk 6 above)
Comment (pacs27) [reply etbrasil/geesebal/timeseries_bcj.py:407]: Ignore the comments and delete them. The functionality is the same. I was trying to solve a problem in another way but finally decided to leave the code as close as possible to the previous one

File: etbrasil/geesebal/meteorology.py (hunk 7 above)
Comment (pacs27) [reply etbrasil/geesebal/meteorology.py:169]: Yes

File: etbrasil/geesebal/tools.py (hunk 8 above)
Comment (pacs27) [reply etbrasil/geesebal/tools.py:1]: Oh you are right, it has been formatted. I'll try to explain it as best as possible here

Basically in this file I have eliminated everything that has to do with the brightness temperature since in the new collection they provide us directly the LST

Line 74: LST calculation  is no longer necessary 
Line 87: brt_r is no longer necessary
Line 161: The same
Line 247: LStemp is no longer necessary

File: etbrasil/geesebal/timeseries_bcj.py (hunk 2 above)
Comment (bjonesneu) [reply etbrasil/geesebal/timeseries_bcj.py:337]: Got it - makes sense I will change it back after merging

File: etbrasil/geesebal/timeseries_bcj.py (hunk 3 above)
Comment (bjonesneu) [reply etbrasil/geesebal/timeseries_bcj.py:135]: I'll leave for future debugging.  I was doing this type of thing in client-side code/notebooks, so having it built into the back-end code is great too

File: etbrasil/geesebal/timeseries_bcj.py (hunk 4 above)
Comment (bjonesneu) [reply etbrasil/geesebal/timeseries_bcj.py:196]: same - will leave in

File: etbrasil/geesebal/timeseries_bcj.py (hunk 6 above)
Comment (bjonesneu) [reply etbrasil/geesebal/timeseries_bcj.py:407]: ok
